datasets/conllpp/conllpp.py

```diff
@@ -0,0 +1,229 @@
+                                "B-SBAR",
+                                "I-SBAR",
+                                "B-UCP",
+                                "I-UCP",
+                                "B-VP",
+                                "I-VP",
+                            ]
+                        )
+                    ),
+                    "ner_tags": datasets.Sequence(
+                        datasets.features.ClassLabel(
+                            names=[
+                                "O",
+                                "B-PER",
+                                "I-PER",
+                                "B-ORG",
+                                "I-ORG",
+                                "B-LOC",
+                                "I-LOC",
+                                "B-MISC",
+                                "I-MISC",
+                            ]
+                        )
+                    ),
+                }
+            ),
+            supervised_keys=None,
+            homepage="https://github.com/ZihanWangKi/CrossWeigh",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        urls_to_download = {
+            "train": f"{_URL}{_TRAINING_FILE}",
+            "dev": f"{_URL}{_DEV_FILE}",
+            "test": f"{_URL}{_TEST_FILE}",
+        }
+        downloaded_files = dl_manager.download_and_extract(urls_to_download)
+
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": downloaded_files["train"]}),
+            datasets.SplitGenerator(name=datasets.Split.VALIDATION, gen_kwargs={"filepath": downloaded_files["dev"]}),
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs={"filepath": downloaded_files["test"]}),
+        ]
+
+    def _generate_examples(self, filepath):
+        logging.info("⏳ Generating examples from = %s", filepath)
+        with open(filepath, encoding="utf-8") as f:
+            guid = 0
+            tokens = []
+            pos_tags = []
+            chunk_tags = []
+            ner_tags = []
+            for line in f:
+                if line.startswith("-DOCSTART-") or line == "" or line == "\n":
+                    if tokens:
+                        yield guid, {
+                            "id": str(guid),
+                            "tokens": tokens,
+                            "pos_tags": pos_tags,
+                            "chunk_tags": chunk_tags,
+                            "ner_tags": ner_tags,
+                        }
+                        guid += 1
+                        tokens = []
+                        pos_tags = []
+                        chunk_tags = []
+                        ner_tags = []
+                else:
+                    # conll2003 tokens are space separated
+                    splits = line.split(" ")
+                    assert len(splits) >= 4, line
```

Comment:
Do you need to keep this assert ?

Reply:
It doesn't seem that necessary since the latter lines try to access the fourth element. Do you suggest removing this assert?

Reply:
You can remove it yes